static/app/components/events/interfaces/spans/spanGroupBar.tsx

```diff
@@ -20,62 +17,50 @@ import {
   SpanGroupRowTitleContent,
 } from 'sentry/components/performance/waterfall/rowTitle';
 import {
-  ConnectorBar,
   TOGGLE_BORDER_BOX,
-  TreeConnector,
   TreeToggle,
   TreeToggleContainer,
 } from 'sentry/components/performance/waterfall/treeConnector';
-import {
-  getDurationDisplay,
-  getHumanDuration,
-  toPercent,
-} from 'sentry/components/performance/waterfall/utils';
-import {t} from 'sentry/locale';
+import {toPercent} from 'sentry/components/performance/waterfall/utils';
 import {EventTransaction} from 'sentry/types/event';
 import {defined} from 'sentry/utils';
-import theme from 'sentry/utils/theme';
 
 import * as DividerHandlerManager from './dividerHandlerManager';
 import * as ScrollbarManager from './scrollbarManager';
 import SpanBarCursorGuide from './spanBarCursorGuide';
 import {MeasurementMarker} from './styles';
-import {EnhancedSpan, ProcessedSpanType, TreeDepthType} from './types';
+import {EnhancedSpan, ProcessedSpanType} from './types';
 import {
   getMeasurementBounds,
   getMeasurements,
-  getSpanGroupBounds,
-  getSpanGroupTimestamps,
-  getSpanOperation,
-  isOrphanSpan,
-  isOrphanTreeDepth,
   SpanBoundsType,
   SpanGeneratedBoundsType,
-  unwrapTreeDepth,
 } from './utils';
 
 const MARGIN_LEFT = 0;
 
 type Props = {
-  continuingTreeDepths: Array<TreeDepthType>;
   event: Readonly<EventTransaction>;
   generateBounds: (bounds: SpanBoundsType) => SpanGeneratedBoundsType;
+  renderGroupSpansTitle: () => React.ReactNode;
+  renderSpanRectangles: () => React.ReactNode;
+  renderSpanTreeConnector: () => React.ReactNode;
   span: Readonly<ProcessedSpanType>;
   spanGrouping: EnhancedSpan[];
   spanNumber: number;
   toggleSpanGroup: () => void;
   treeDepth: number;
 };
 
-class SpanGroupBar extends React.Component<Props> {
-  renderGroupedSpansToggler() {
-    const {spanGrouping, treeDepth, toggleSpanGroup} = this.props;
+export default function SpanGroupBar(props: Props) {
+  function renderGroupedSpansToggler() {
```

Comment:
True, but that was an oversight on my part since these functions previously were within a class based component, so the functions being defined within the component were not an issue there. I will factor these functions out of the component so that they aren't recreated upon re-renders